TFUtil.py

```diff
@@ -6667,3 +6667,32 @@ def get_positional_encoding(num_channels, length=None, position=None, min_timesc
   signal = tf.concat([tf.sin(scaled_time), tf.cos(scaled_time)], axis=1)
   signal = tf.pad(signal, [[0, 0], [0, num_channels % 2]])  # (length, channels)
   return signal
+
+def check_contains_only_det_ops(print_trace=False):
+  """
+  :param bool print_trace: prints a trace of which ops are checked
+  :return:  are all ops used on the available devices deterministic?
+  """
+  # List of non deterministic (NDT) ops (depending on version and device)
+  non_deterministic_ops = {
+    "Mean": ["1.5.0", ["GPU"]]
+    # "Reshape": ["1.13.0", ["CPU", "GPU"]]  # Not NDT just here for test purposes
+  }
+
+  # Getting all available device types
+  device_types = [type.device_type for type in get_tf_list_local_devices()]
```

Comment:
Replace `type` (bad variable name for the device, also shadows builtin function) by `dev`. (Btw, do you use PyCharm? It will warn you about things like this.)